test/object-store/sync/client_reset.cpp

```diff
@@ -1912,20 +1912,21 @@ TEST_CASE("sync: Client reset during async open", "[sync][pbs][client reset][baa
     };
 
     auto realm_task = Realm::get_synchronized_realm(realm_config);
-    auto realm_pf = util::make_promise_future<SharedRealm>();
+    auto realm_pf = util::make_promise_future<ThreadSafeReference>();
     realm_task->start([&](ThreadSafeReference ref, std::exception_ptr ex) {
         try {
             if (ex) {
                 std::rethrow_exception(ex);
             }
-            auto realm = Realm::get_shared_realm(std::move(ref));
-            realm_pf.promise.emplace_value(std::move(realm));
+            realm_pf.promise.emplace_value(std::move(ref));
```

Comment:
I think this change makes it fail to test the thing it's supposed to be testing. Opening the Realm inside the async open callback is the more complicated scenario that's more likely to have problems.